server/src/main/java/org/elasticsearch/action/admin/indices/create/TransportCreateIndexAction.java

```diff
@@ -187,14 +187,20 @@ private static CreateIndexClusterStateUpdateRequest buildSystemIndexUpdateReques
             aliases = Set.of(new Alias(descriptor.getAliasName()).isHidden(true).writeIndex(true));
         }
 
-        // Throw an error if we are trying to directly create a system index other than the primary system index (or the alias)
-        if (request.index().equals(descriptor.getPrimaryIndex()) == false && request.index().equals(descriptor.getAliasName()) == false) {
+        boolean indexMigrationInProgress = request.index().endsWith(SystemIndices.UPGRADED_INDEX_SUFFIX);
```

Comment:
Does this actually indicate that the migration is in progress?  If this index exists, doesn't this only show that the migration has at some point started (and perhaps even completed)? Or, is it not possible to get to this point if the migraiton has already completed?